flockoff/validator/database.py

```
@@ -1,57 +1,123 @@
+            c = self.conn.cursor()
+            c.execute("UPDATE miner_scores SET score = ? WHERE uid = ?", (new_score, uid))
+            if c.rowcount == 0:
+                # row absent – insert with unknown hotkey
+                c.execute(
+                    "INSERT INTO miner_scores (uid, hotkey, score) VALUES (?, '', ?)",
+                    (uid, new_score),
+                )
+            self.conn.commit()
+        except sqlite3.Error as e:
+            logger.error(f"Failed to update score for UID {uid}: {str(e)}")
+            raise DatabaseError(f"Failed to update score: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)